query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryRequestUtil.java

```diff
@@ -139,4 +139,20 @@ public static String getLogicalColumnName(Expression expression) {
                 + " expression type only");
     }
   }
+
+  public static String getAlias(Expression expression) {
```

Comment:
would recommend making this something like getResultName and putting the fallback logic (alias -> column name) in here. In the gateway code I wrote, those two utils - to resolve an attribute ID out of an expression and to resolve a result name out of an expression were used all over the place. It's also helpful to return an optional and let the caller decide whether to throw if missing or not, as you saw in the isMatchingFilter case where you needed to change the logic.

Reply:
(from gateway code, as an example)
```java
  public static Optional<String> getSelectionResultName(Expression expression) {
    switch (expression.getValueCase()) {
      case COLUMNIDENTIFIER:
        return Optional.of(
            expression.getColumnIdentifier().getAlias().isEmpty()
                ? expression.getColumnIdentifier().getColumnName()
                : expression.getColumnIdentifier().getAlias());
      case ATTRIBUTE_EXPRESSION:
        return Optional.of(
            expression.getAttributeExpression().getAlias().isEmpty()
                ? expression.getAttributeExpression().getAttributeId()
                : expression.getAttributeExpression().getAlias());
      case FUNCTION:
        FunctionExpression functionExpression = expression.getFunction();
        if (!functionExpression.getAlias().isEmpty()) {
          return Optional.of(functionExpression.getAlias());
        }
        String argumentString =
            functionExpression.getArgumentsList().stream()
                .map(ExpressionReader::getSelectionResultName)
                .flatMap(Optional::stream)
                .collect(Collectors.joining(","));

        return Optional.of(
            String.format("%s_%s", functionExpression.getFunction(), argumentString));
      default:
        return Optional.empty();
    }
  }
```

Reply:
Is optional required since it will always be present ?

Reply:
It won't always be present, we don't know what's in the expression. It could be a literal, for example. Optional is preferable to throwing the exception, because it makes it easier for the caller to decide what to do if it's missing - throw a specific exception, ignore etc.

Reply:
got it.

Reply:
resolved